src/util/fs.js

```diff
@@ -527,6 +552,15 @@ export function copy(src: string, dest: string, reporter: Reporter): Promise<voi
   return copyBulk([{src, dest}], reporter);
 }
 
+const copyFileWithAttributes = async function(data: CopyFileAction, cleanup: Function): Promise<void> {
```

Comment:
This is a terrible name, I should probably change this to `safeCopyFile` with an explanation of `unlink` and `cleanup` parts. What do you think?

Reply:
The name looks fine, but a comment to explain the difference with copyFile would be useful.

I also feel like the arguments are also kinda obfuscated because of the custom CopyFileAction type.